samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/model/DataTableConfig.java

```diff
@@ -0,0 +1,135 @@
+package org.sakaiproject.tool.assessment.ui.model;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.LinkedHashSet;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.ResourceBundle;
+import java.util.Set;
+
+import lombok.Builder;
+import lombok.Data;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+@Data
+@Builder
+public class DataTableConfig {
+
+
+    private static final String SORT_ASCENDING = "asc";
+    private static final String SORT_DESCENDING = "desc";
+    private static final String I18N_ALL = "all";
+    private static final String I18N_ARIA = "aria";
+    private static final String I18N_EMPTY_TABLE = "emptyTable";
+    private static final String I18N_INFO = "info";
+    private static final String I18N_INFO_EMPTY = "infoEmpty";
+    private static final String I18N_INFO_FILTERED = "infoFiltered";
+    private static final String I18N_LENGTH_MENU = "lengthMenu";
+    private static final String I18N_NEXT = "next";
+    private static final String I18N_PAGINATE = "paginate";
+    private static final String I18N_PREVIOUS = "previous";
+    private static final String I18N_SEARCH = "search";
+    private static final String I18N_SORT_ASCENDING = "sortAscending";
+    private static final String I18N_SORT_DESCENDING = "sortDescending";
+    private static final String I18N_ZERO_RECORDS = "zeroRecords";
+
+
+    public static class DataTableConfigBuilder {
+
+        public DataTableConfigBuilder initialSort(int columnIndex, boolean ascending) {
+            order = new HashSet<>() {{
+                    add(new HashSet<>() {{
+                            add(5);
+                    }});
+                    add(new HashSet<>() {{
+                            add(ascending ? SORT_ASCENDING : SORT_DESCENDING);
+                    }});
+            }};
+
+            return this;
+        }
+
+        public DataTableConfigBuilder lengthMenuMessage(String lengthMenuMessage) {
+            addMessage(I18N_LENGTH_MENU, lengthMenuMessage);
+
+            return this;
+        }
+
+        public DataTableConfigBuilder filterMessage(String filterMessage) {
+            addMessage(I18N_INFO_FILTERED, filterMessage);
+
+            return this;
+        }
+
+        public DataTableConfigBuilder emptyMessage(String emptyMessage) {
+            addMessage(I18N_INFO_EMPTY, emptyMessage);
+            addMessage(I18N_EMPTY_TABLE, emptyMessage);
+            addMessage(I18N_ZERO_RECORDS, emptyMessage);
+
+            return this;
+        }
+
+        private void addMessage(String messageKey, String message) {
+            if (language == null) {
+                language = new HashMap<>();
+            }
+
+            language.put(messageKey, message);
+        }
+    }
+
+
+    private Boolean paging;
```

Comment:
Moving these variables to the top?

Reply:
Sure, makes sense. I moved the builder down :+1: